### DISCUSSION TITLE
Shall a renderer support multiple processors?

### DISCUSSION THREAD

**Discussion by jakubjezek001** [ynput/LabLib#12] [Q&A]
In regards to supporting multiple processors, it is worth considering if this would be a beneficial step. Potential issues could arise with execution order, and separation of processors must be carefully considered, possibly through inheritance. Additionally, another concern is the potential complexity that may come with utilizing multiple processors, which could make the workflow more challenging to manage. We should carefully weigh the pros and cons before committing to this approach and explore alternatives that may better suit our needs without adding unnecessary complexity. Ultimately, while multiple processor support may be tempting, thorough evaluation is necessary to ensure positive impact on our workflow and project. Let us keep these points in mind and have a thoughtful discussion before proceeding.

**Comment by tweak-wtf** [comment]
thing is that i can't really image a use case where someone would like to render multiple processors. e.g. rendering an `AYONOCIOLookFileProcessor` and a `AYONHieroEffectsFileProcessor`.
also i'm thinking about what would happen if a user passes in processors that cancel eachother out or simply cannot be render in a single `oiiotool` command. i believe figuring out which processors are compatible with each other might turn out to be a pita.

the exception to this would be `BurninProcessor` and `SlateProcessor`. with these u do want to have, eg. a OCIO transform with burnins and a slate.
however here we could debate whether or not burnins and slates should be `renderers` and not `processors` .
Currently i'm supporting burnins with the `--text` flag of `oiiotool` but that might change

**Comment by maxpareschi** [comment]
Well, this is probably true.
The expected usage I had in mind would have been a StandardRenderer, a ReviewRenderer, a SlateRenderer and a BurninRenderer. That being said, for speedy processing (especially when you discard intermediated after use) the chain should be done as much as possible in one call (which oiio tool supports at the expense of memory)

This is probably something worth considering, since what I was planning was to have a generalized framework that can be extended as needed, not a single end product.

**Reply by tweak-wtf** [reply]
Hey Max, great to see u here :)
so far it worked quite well to concatenate all the args for color and repositioning in on `oiiotool` command including basic burnins.
The custom slater and burnin renderer would be the first case where there is a `pre-oiio` processing, meaning a phase where burnins (with alpha) and the slate should be rendered, then color managed through `oiiotool` and in the end get optionally transcoded to some container using `ffmpeg`.
feel free to check out https://github.com/ynput/LabLib/pull/8 for current renderer features
